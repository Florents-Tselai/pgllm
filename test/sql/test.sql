select jsonb_llm_generate('{"message": "hello", "k1": "v1", "k2": "v2", "k3": 3}'::jsonb, 'no-existing-model', '{"n": 3}'::jsonb);

select jsonb_llm_generate('{"message": "hello", "k1": "v1", "k2": "v2", "k3": 3}'::jsonb, 'repeat-3', '{"n": 3}'::jsonb);

select jsonb_llm_generate('{"message": "hello", "a": [1, "one", 2, 3]}'::jsonb, 'pyupper', '{"n": 3}'::jsonb);

select jsonb_llm_generate('{"message": "hello"}'::jsonb, 'pyupper', '{"n": 3}'::jsonb);

select llm_generate('hello'::text, 'repeat-3'::text, '{"n": 3}'::jsonb);

select jsonb_llm_embed('"hello"'::jsonb, 'repeat');
