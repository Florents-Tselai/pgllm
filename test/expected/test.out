select jsonb_llm_generate('{"message": "hello", "k1": "v1", "k2": "v2", "k3": 3}'::jsonb, 'no-existing-model', '{"n": 3}'::jsonb);
ERROR:  pgllm: model no-existing-model is not supported

select jsonb_llm_generate('{"message": "hello", "k1": "v1", "k2": "v2", "k3": 3}'::jsonb, 'repeat-3', '{"n": 3}'::jsonb);
                         jsonb_llm_generate                          
---------------------------------------------------------------------
 {"k1": "V1", "k2": "V2V2V2", "k3": 3, "message": "HELLOHELLOHELLO"}
(1 row)

select jsonb_llm_generate('{"message": "hello", "a": [1, "one", 2, 3]}'::jsonb, 'pyupper', '{"n": 3}'::jsonb);
             jsonb_llm_generate              
---------------------------------------------
 {"a": [1, "ONE", 2, 3], "message": "HELLO"}
(1 row)

select jsonb_llm_generate('{"message": "hello"}'::jsonb, 'pyupper', '{"n": 3}'::jsonb);
  jsonb_llm_generate  
----------------------
 {"message": "HELLO"}
(1 row)

select llm_generate('hello'::text, 'repeat-3'::text, '{"n": 3}'::jsonb);
   llm_generate    
-------------------
 "HELLOHELLOHELLO"
(1 row)

select jsonb_llm_embed('"hello"'::jsonb, 'repeat');
 jsonb_llm_embed 
-----------------
 "hello"
(1 row)

